### PR TITLE
Attendance Overview tab in Child Details should hide archived 

### PR DESCRIPTION
**Pull request by Abhinegi2**
closes: #2587

### Visible/Frontend Changes

- [x] Added Previous Activities tab from where user can view historical recurring activities based on selected activity

### CODE REVIEW

File: src/app/child-dev-project/children/child-details/grouped-child-attendance/grouped-child-attendance.component.html [hunk 1]
```diff
@@ -10,4 +10,32 @@
...
+        <mat-option
+          *ngFor="let archiveActivity of archiveActivities"
+          [value]="archiveActivity.title"
+        >
+          {{ archiveActivity.title }}
+        </mat-option>
+      </mat-select>
+    </mat-form-field>
+    <div *ngIf="selectedActivity">
+      <div *ngFor="let seletcedArchiveActivity of seletcedArchiveActivities">
```
Comment (sleidig) [inline src/app/child-dev-project/children/child-details/grouped-child-attendance/grouped-child-attendance.component.html:32]: typo?

File: src/app/child-dev-project/children/child-details/grouped-child-attendance/grouped-child-attendance.component.ts [hunk 2]
```diff
@@ -39,9 +44,31 @@ export class GroupedChildAttendanceComponent implements OnInit {
...
+      this.entity.getId(),
+    );
+
+    this.activities = allActivities.filter(
+      (a) =>
+        !a.excludedParticipants.includes(this.entity.getId()) &&
+        a.isActive == true,
+    );
+
+    this.archiveActivities = allActivities.filter(
```
Comment (sleidig) [inline src/app/child-dev-project/children/child-details/grouped-child-attendance/grouped-child-attendance.component.ts:57]: let's call this "archivedActivities"

File: src/app/child-dev-project/children/child-details/grouped-child-attendance/grouped-child-attendance.component.html [hunk 3]
```diff
@@ -10,4 +10,32 @@
...
+      This section displays historical activities in which the student was
+      previously enrolled.
+    </p>
+
+    <mat-form-field>
+      <mat-label>Select a Previous Activity</mat-label>
+      <mat-select (selectionChange)="onActivityChange($event.value)">
+        <mat-option
+          *ngFor="let archiveActivity of archiveActivities"
+          [value]="archiveActivity.title"
```
Comment (sleidig) [inline src/app/child-dev-project/children/child-details/grouped-child-attendance/grouped-child-attendance.component.html:25]: probably better if you set the object as value directly (ˋ[value]="archiveActivity"ˋ). Then you could also simply do ˋ(selectionChange)="selectedArchivedActivity=$event.value"ˋ above (and replace the array of selected activities + boolean selectedActivity with simply one variable storing the currently selected one)

**Review by sleidig — CHANGES_REQUESTED** [review]
Nice work, just some minor code cleanup remarks

File: src/app/child-dev-project/children/child-details/grouped-child-attendance/grouped-child-attendance.component.html [hunk 4]
```diff
@@ -10,4 +10,32 @@
     >
     </app-activity-attendance-section>
   </mat-tab>
+
+  <mat-tab *ngIf="archivedActivities.length > 0" label="Previous Activities">
+    <p i18n>
+      This section displays historical activities in which the student was
+      previously enrolled.
```
Comment (sleidig) [inline src/app/child-dev-project/children/child-details/grouped-child-attendance/grouped-child-attendance.component.html:17]:
```suggestion
      This section displays historical activities that have been archived.
```

File: src/app/child-dev-project/children/child-details/grouped-child-attendance/grouped-child-attendance.component.html [hunk 5]
```diff
@@ -10,4 +10,32 @@
...
+    </p>
+
+    <mat-form-field>
+      <mat-label>Select a Previous Activity</mat-label>
```
Comment (sleidig) [inline src/app/child-dev-project/children/child-details/grouped-child-attendance/grouped-child-attendance.component.html:21]:
```suggestion
      <mat-label i18n>Select activity</mat-label>
```

File: src/app/child-dev-project/children/child-details/grouped-child-attendance/grouped-child-attendance.component.html [hunk 6]
```diff
@@ -10,4 +10,32 @@
...
+        <mat-option
+          *ngFor="let archivedActivity of archivedActivities"
+          [value]="archivedActivity"
+        >
+          {{ archivedActivity.title }}
+        </mat-option>
+      </mat-select>
+    </mat-form-field>
+    <div *ngIf="selectedActivity">
+      <div *ngFor="let selectedArchivedActivity of selectedArchivedActivities">
```
Comment (sleidig) [inline src/app/child-dev-project/children/child-details/grouped-child-attendance/grouped-child-attendance.component.html:32]: how can there be multiple selected activities in the current design?

File: src/app/child-dev-project/children/child-details/grouped-child-attendance/grouped-child-attendance.component.ts [hunk 7]
```diff
@@ -39,9 +44,31 @@ export class GroupedChildAttendanceComponent implements OnInit {
...
   }
+
+  onActivityChange(selectedArchivedActivity: RecurringActivity) {
+    this.selectedActivity = true;
+    this.selectedArchivedActivities = this.archivedActivities.filter(
+      (a) =>
+        a.title == selectedArchivedActivity.title &&
+        !a.excludedParticipants.includes(this.entity.getId()),
+    );
+  }
```
Comment (sleidig) [inline src/app/child-dev-project/children/child-details/grouped-child-attendance/grouped-child-attendance.component.ts:73]: see my other comment, please don't rely on the title as there may be duplicates.

I suggest
```suggestion
    this.selectedActivity = selectedArchivedActivity;
  }
```

(with the relevant type and template changes, of course)

**Review by sleidig — CHANGES_REQUESTED** [review]

**Comment by sleidig** [comment]
Functionality works well 👍

File: src/app/child-dev-project/children/child-details/grouped-child-attendance/grouped-child-attendance.component.ts (hunk 7 above)
Comment (Abhinegi2) [reply src/app/child-dev-project/children/child-details/grouped-child-attendance/grouped-child-attendance.component.ts]: updated the template and ts file

**Review by sleidig — APPROVED** [review]

**Pull request merged**